qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java

```diff
@@ -911,7 +913,7 @@ public void testSnapshotRestore() throws IOException {
 
         // In 7.0, type names are no longer expected by default in put index template requests.
         // We therefore use the deprecated typed APIs when running against the current version, but testing with a pre-7 version
-        if (isRunningAgainstOldCluster() == false && getOldClusterVersion().major < 7) {
+        if (isRunningAgainstOldCluster() == false && templateUsesTypes) {
```

Comment:
Is this just a refactor, or is this a necessary part of the fix as well? I'm having some trouble understanding.

Reply:
Yes, just a refactor to make the whole version juggling easier to read. I introduced the flag earlier if the type is added to the template. I had some trouble understanding the code without it and found it worth being explicit here.

Reply:
👍